include/cucumber-cpp/internal/step/StepManager.hpp

```diff
@@ -62,6 +62,11 @@ class InvokeArgs {
 
     template<class T> T getInvokeArg(size_type i) const;
     const Table & getTableArg() const;
+
+    size_type argsSize() const {
+      return args.size();
+    }
+
```

Comment:
I'd say it belongs to a PR of its own (but it's hard to know if it makes sense without knowing what it should be used for) or it should live in your own branch. It didn't make any sense fixing the compiler warnings or in the OSX build.